### PR TITLE
Typetest

### PR DESCRIPTION
**Pull request by andresgalante**
@srambach @jgiardino @cshinn This PR introduces a typography template:
http://blog.andresgalante.com/patternfly-atomic/?p=templates-typography

It also adds variables and utilities for shadows:
http://blog.andresgalante.com/patternfly-atomic/?p=basics-box-shadow

And it updates the guidelines to cover spacing, shadows and gradients.

What do you think?

### CODE REVIEW

File: CODE-GUIDELINES.md [hunk 1]
```diff
@@ -595,6 +598,100 @@ Never use `!important` to raise the specificity of a rule. In well architected C
...
+
+```html
+<p class="mb-lg">This paragraph has a large bottom margin</p>
+```
+
+<!-- ============================================================ -->
+
+## Shadows
+
+PatternFly has 4 levels of shadows, always variables to apply `box-shadow`:
```
Comment (srambach) [inline CODE-GUIDELINES.md]: Maybe just say there are 6 types? Is the term "levels" used elsewhere (besides in the template)?

File: CODE-GUIDELINES.md [hunk 2]
```diff
@@ -595,6 +598,100 @@ Never use `!important` to raise the specificity of a rule. In well architected C
...
+```html
+<p class="pf-box-shadow-md">This paragraph has a medium box shadow</p>
+
+<button class="btn btn-primary pf-box-shadow-none">This button has no shadow</button>
+```
+
+<!-- ============================================================ -->
+
+## Gradients
+PatternFly has one slight gradient to define panel headers, table headers, secondary btns and other elements. Always use the PatternFly gradient mixing to define it.
```
Comment (srambach) [inline CODE-GUIDELINES.md]: Typo "mixing" for "mixin"

**Comment by andresgalante** [comment]
@srambach thanks 😄  I've fixed it.

**Comment by srambach** [comment]
Looks great, @andresgalante !

File: CODE-GUIDELINES.md [hunk 3]
```diff
@@ -595,6 +598,100 @@ Never use `!important` to raise the specificity of a rule. In well architected C
...
+
+```sass
+thead th {
+  @include pf-gradient;
+}
+```
+
+<!-- ============================================================ -->
+
+## Box Shadow
```
Comment (jgiardino) [inline CODE-GUIDELINES.md]: Did you mean to keep this section here?

**Comment by jgiardino** [comment]
This looks great!

**Comment by jgiardino** [comment]
bitmoji

**Comment by andresgalante** [comment]
@jgiardino thanks! I've removed it.

**Pull request merged**